lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs

```diff
@@ -78,7 +78,8 @@ public Task<FinalState> GetJobFinalStatus(string appId)
         public void Submit(JobRequest jobRequest)
         {
             var configModule = AzureBatchClientConfiguration.ConfigurationModule;
-            string jobId = jobRequest.JobIdentifier;
+            var timestamp = DateTime.Now.ToString("yyyyMMddHHmmssfff");
+            string jobId = string.Join("-", jobRequest.JobIdentifier, timestamp);
```

Comment:
One thing to keep in mind is that there is a limit to how long Azure Batch job ID can be. So if jobRequest.JobIdentifier is too long, we might exceed it.

Reply:
Updated as what Java did. For local tmp folder, it is still doing with timestamp for user to check.

Reply:
fixed.